pkg/cidata/cidata.TEMPLATE.d/boot/04-persistent-data-volume.sh

```diff
@@ -61,19 +61,34 @@ if [ "$(awk '$2 == "/" {print $3}' /proc/mounts)" == "tmpfs" ]; then
 				PART=$(lsblk --list /dev/"${DISK}" --noheadings --output name,type | awk '$2 == "part" {print $1}')
 				mkfs.ext4 -L data-volume /dev/"${PART}"
 				mount -t ext4 /dev/disk/by-label/data-volume /mnt/data
-				# Unmount all mount points under /tmp so we can move it to the data volume:
-				# "mount1 on /tmp/lima type 9p (rw,dirsync,relatime,mmap,access=client,trans=virtio)"
-				for MP in $(mount | awk '$3 ~ /^\/tmp\// {print $3}'); do
-					umount "${MP}"
-				done
 				# setup apk package cache
 				mkdir -p /mnt/data/apk/cache
 				mkdir -p /etc/apk
 				ln -s /mnt/data/apk/cache /etc/apk/cache
+				# Prepare /mnt.tmp.sh (used for restoring mounts later)
+				echo "#!/bin/sh" >/mnt.tmp.sh
```

Comment:
Don't you have to put the mount script on `/mnt/data` so it is still there after a reboot?

On the other hand, I don't remember why the remounting has to be done after a reboot; it should only be needed on first boot when mounts have been unmounted.

Reply:
> I don't remember why the remounting has to be done after a reboot;

Oh, it has to be done after the persistent directories have been bind-mounted, so the host directories are not hidden behind.

Reply:
This is a tmp script and removed at the end

Reply:
It is currently outside the first-boot branch:

```shell
if [ "$(awk '$2 == "/" {print $3}' /proc/mounts)" == "tmpfs" ]; then
  if [ -e /dev/disk/by-label/data-volume ]; then
    …
  else
    # create /mnt.tmp.sh
  fi
  # bind-mount all persistent directories
  /mnt.tmp.sh
fi
```

The whole outer conditional is somewhat pointless, as it will always be true as long as we boot from an ISO. But the code after the inner conditional is executed on every boot.

Reply:
fixed to retain it under /mnt/data